_gsocproposals/2022/proposal_PODIO-Julia,md

```diff
@@ -0,0 +1,90 @@
+---
+title: Interfacing PODIO to Julia
+layout: gsoc_proposal
+project: HSF
+year: 2022
+difficulty: medium
+duration: 175
+mentor_avail: June-September
+organization:
+    - CERN
+    - DESY
+---
+
+# Description
+
+<div style="text-align:center; padding:25px; float:right">
+<img src="{{ '/images/others/julia.png' | relative_url }}" alt="Julia Logo" width="100px" />
+</div>
+
+Currently in high-energy physics (HEP) there are [two main languages
+used][SCRoundtable]: **C++** for numerically intensive code, where execution
+speed is paramount; and **Python**, for interactivity and ease of development
+(often used as 'glue' between high-performance code modules). Recently there has
+been [increasing interest][SCRoundtable] in [**Julia**][julialang] as an
+alternative/additional language for HEP. This could offer the convenience
+features of Python, but the optimal runtime speed of C++
+([Julia in HEP Paper][JuliaHEPPaper], [Julia HEP Github][JHEPGH]).
+
+In order to continue this investigation, this project will interface the data model library
+PODIO with Julia. This will allow to read existing data files into a Julia program.
+
+##  PODIO
+
+PODIO is a data-model library based on the idea of representing information as plain-old-data (PODs) 
+instead of using a deep object-oriented model. This allows for better performance and interoperability.
+Data models are defined by a dedicated YAML-syntax and the corresponding code is auto-generated. So far 
+both C++ and Python are supported.
+
+## Task ideas and expected results
+
+The task is to create a code generator to add support for Julia as end-user language, and to carry out 
```

Comment:
Maybe we could specify that "the code generator" in this case would involve mainly some new jinja templates as the main infrastructure for code generation is already in place. But for the project proposal that might be too detailed. I am also fine with the current formulation.

Reply:
thanks. I think we can leave it as is for now.